IDE/Renesas/e2studio/RX72N/EnvisionKit/wolfssl_demo/wolfssl_tsip_unit_test.c

```diff
@@ -0,0 +1,1020 @@
+ *
+ * Copyright (C) 2006-2022 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfssl/wolfcrypt/sha.h>
+#include <wolfssl/wolfcrypt/sha256.h>
+#include <wolfssl/wolfcrypt/aes.h>
+#include <wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h>
+
+#include "FreeRTOS.h"
+#if defined(FREERTOS_TCP)
+    #include "FreeRTOS_IP.h"
+    #include "FreeRTOS_Sockets.h"
+    #include "platform/iot_network.h"
+    #include "platform.h"
+#endif
+
+#ifndef NO_SHA
+ int sha_test();
+#endif
+
+#ifndef NO_SHA256
+ int sha256_test();
+#endif
+
+#define SMALL_STACK_SIZE (1 * 1024)
+#define STACK_SIZE       (4 * 1024)
+
+#define RESULT_STR(ret)  if (ret == 0)\
+                          printf(" passed \n");\
+                         else \
+                          printf(" failed \n");
+
+static xSemaphoreHandle exit_semaph;
+static byte exit_loop = 0;
+static byte sha_multTst_rslt = 0;
+static byte sha256_multTst_rslt = 0;
+static byte Aes128_Cbc_multTst_rslt = 0;
+static byte Aes256_Cbc_multTst_rslt = 0;
+static byte Aes128_Gcm_multTst_rslt = 0;
+static byte Aes256_Gcm_multTst_rslt = 0;
+
+#if defined(HAVE_AES_CBC)
+
+#if defined(WOLFSSL_AES_128)
+ tsip_aes_key_index_t g_user_aes128_key_index1;
+ tsip_aes_key_index_t g_user_aes128_key_index2;
+#endif
+
+#if defined(WOLFSSL_AES_256)
+ tsip_aes_key_index_t g_user_aes256_key_index1;
+ tsip_aes_key_index_t g_user_aes256_key_index2;
+#endif
+
+#endif
+
+typedef struct tagInfo
+{
+    tsip_aes_key_index_t aes_key;
+} Info;
+
+#if defined(HAVE_AES_CBC) && defined(WOLFSSL_AES_128)
+
+static int tsip_aes_cbc_test(int prnt, tsip_aes_key_index_t* aes_key)
+{
+
+    Aes  aes[1];
+
+    byte cipher[AES_BLOCK_SIZE];
+    byte plain[AES_BLOCK_SIZE];
+    int  ret = 0;
+
+    WOLFSSL_SMALL_STACK_STATIC const byte msg[] = { 
+        /* "Now is the time for all " w/o trailing 0 */
+        0x6e,0x6f,0x77,0x20,0x69,0x73,0x20,0x74,
+        0x68,0x65,0x20,0x74,0x69,0x6d,0x65,0x20,
+        0x66,0x6f,0x72,0x20,0x61,0x6c,0x6c,0x20
+    };
+    byte key[] = "0123456789abcdef   ";  /* align */
+    byte iv[]  = "1234567890abcdef   ";  /* align */
+
+    XMEMSET(cipher, 0, AES_BLOCK_SIZE);
```

Comment:
We can use ForceZero() instead of XMEMSET( .., 0, .. ).

Reply:
Fixed. All XMEMSET at the file are replaced by ForceZero()